akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -289,7 +289,10 @@ private[http] object HttpServerBluePrint {
         def onPush(): Unit = {
           openTimeouts.head.clear()
           openTimeouts = openTimeouts.tail
-          push(responseOut, grab(responseIn))
+
+          // it may have been completed by a timeout already
```

Comment:
only out was cancelled, completeStage effectively cancels both which actually may be what we need here - testing currently.